docs/deployment/gdal_config.md

```diff
@@ -0,0 +1,129 @@
+# GDAL Configuration
+
+Titiler makes use of several great underlying libraries, including [GDAL][gdal]
+and [Python bindings to GDAL][rasterio]. An effective deployment of titiler
+generally requires tweaking GDAL configuration settings. This document provides
+an overview of relevant settings. Full documentation from GDAL is available
+[here][gdal_config_docs].
+
+[gdal]: https://gdal.org/
+[rasterio]: https://rasterio.readthedocs.io/
+[gdal_config_docs]: https://trac.osgeo.org/gdal/wiki/ConfigOptions
+
+## Setting a config variable
+
+GDAL configuration is modified using environment variables. Thus in order to
+change a setting you'll need to set environment variables through your
+deployment mechanism. For example, in order to test locally you'd set an
+environment variable in bash:
+
+```bash
+export GDAL_HTTP_MULTIPLEX=YES
+```
+
+## Available configuration settings
+
+#### `GDAL_HTTP_MULTIPLEX`
```

Comment:
Maybe we could order the section in a kind of `priority`. I usually set HTTP_MULTIPLEX but I
 m 99.9% that it's useless because no cloud provider storage supports http2 for now. 

IMO the order could be :
- `GDAL_HTTP_MERGE_CONSECUTIVE_RANGES`
- `GDAL_DISABLE_READDIR_ON_OPEN`
- `CPL_VSIL_CURL_ALLOWED_EXTENSIONS`
- `GDAL_INGESTED_BYTES_AT_OPEN`
- `GDAL_CACHEMAX`
- `VSI_CACHE`
- `AWS_REQUEST_PAYER`
- Other cloud provider env
- PROJ_NETWORK (only for goal 3.2 / proj 7.1)

Reply:
Yeah I was def going to ask what you thought the order should be. I just kinda threw ideas down randomly. Also maybe would be good to rename the doc to "config" with a GDAL section, and then a separate section for AWS with `AWS_REQUEST_PAYER`.

Reply:
👍